h/admin.py

```diff
@@ -250,6 +273,31 @@ def groups_index_csv(request):
 
     return {'header': header, 'rows': rows}
 
+class UserDeletionError(Exception):
+    pass
+
+def delete_user(request, user):
+    """
+    Deletes a user with all their group memberships and annotations.
+
+    Raises UserDeletionError when deletion fails with the appropriate error
+    message.
+    """
+
+    for group in user.groups:
+        if group.creator == user:
+            raise UserDeletionError('Cannot delete user who is a group creator.')
```

Comment:
Ah, now I think about it, there's a slight problem with this, which is that at the moment it's possible for someone to be a creator of a group that they've subsequently left. That is, they can be a creator of a group that isn't in `user.groups`.

So the condition probably wants to be `Group.query.filter(Group.creator == user).count() > 0`.